cluster/cluster.go

```diff
@@ -22,108 +26,104 @@ func validMode(m string) bool {
 }
 
 var (
-	ThisNode *Node
-	Mode     ModeType
-	shutdown = make(chan struct{})
-
-	mu    sync.Mutex
-	peers = make([]*Node, 0)
+	Mode           ModeType
+	Manager        *ClusterManager
+	cfg            *memberlist.Config
+	clusterPrimary = stats.NewBool("cluster.primary")
 )
 
-func Init(name, version string, started time.Time) {
-	ThisNode = &Node{
-		name:          name,
-		started:       started,
-		version:       version,
-		primary:       false,
-		primaryChange: time.Now(),
-		stateChange:   time.Now(),
+func Init(name, version string, started time.Time, scheme string, port int) {
```

Comment:
terms like `port`, `scheme`, `ListenPort` and `ListenScheme` are getting ambiguous.

it looks like MT will listen in two ways: for its api (http), and also for gossip (tcp  and udp).
I propose we use terms like apiPort, apiScheme (instead of listen*) in the code next to clusterPort and clusterHost. so that when you look any any scheme/port/addr variable, it's more clear which one it is.